psi4/driver/driver_findif.py

```diff
@@ -0,0 +1,798 @@
+                offdiag_row = offdiag_energies[offdiag_row_index]
+                if data["num_pts"] == 3:
+                    fc = (+offdiag_row[0] + offdiag_row[1] + 2 * ref_energy - energies[i][0] - energies[i][1] -
+                          energies[j][0] - energies[j][1]) / (2 * data["disp_size"]**2)
+                elif data["num_pts"] == 5:
+                    fc = (-offdiag_row[0] - offdiag_row[1] + 9 * offdiag_row[2] - offdiag_row[3] - offdiag_row[4] +
+                          9 * offdiag_row[5] - offdiag_row[6] - offdiag_row[7] + energies[i][0] - 7 * energies[i][1] -
+                          7 * energies[i][2] + energies[i][3] + energies[j][0] - 7 * energies[j][1] -
+                          7 * energies[j][2] + energies[j][3] + 12 * ref_energy) / (12 * data["disp_size"]**2)
+                H_irr[i, j] = fc
+                H_irr[j, i] = fc
+                offdiag_row_index += 1
+
+        B_pi.append(data["salc_list"].matrix_irrep(h))
+        H_pi.append(_process_hessian_symmetry_block(H_irr, B_pi[-1], massweighter, irrep_lbls[h], data["print_lvl"]))
+
+    # All blocks of the Hessian are now constructed!
+    return _process_hessian(H_pi, B_pi, massweighter, data["print_lvl"])
+
+
+def gradient_from_energy_geometries(molecule):
+    """
+    Generate geometries for a gradient by finite difference of energies.
+    
+    Parameters
+    ----------
+    molecule : qcdb.molecule or psi4.core.Molecule
+        The molecule to compute the gradient of.
+
+    Returns
+    -------
+    disp_geoms : list of psi4.core.Matrix
+        A list of displaced geometries to compute energies at.
+
+    Notes
+    -----
+    Only symmetric displacements are necessary, so user specification of
+    symmetry is disabled.
+    """
+    return _geom_generator(molecule, -1, "1_0")
+
+
+def hessian_from_gradient_geometries(molecule, irrep):
+    """
+    Generate geometries for a hessian by finite difference of energies.
+    
+    Parameters
+    ----------
+    molecule : qcdb.molecule or psi4.core.Molecule
+        The molecule to compute the frequencies of.
+    irrep : int
+        The Cotton ordered irrep to get frequencies for. Choose -1 for all
+        irreps.
+
+    Returns
+    -------
+    disp_geoms : list of psi4.core.Matrix
+        A list of displaced geometries to compute gradients at.
+    """
+    return _geom_generator(molecule, irrep, "2_1")
+
+
+def hessian_from_energy_geometries(molecule, irrep):
+    """
+    Generate geometries for a hessian by finite difference of energies.
+    
+    Parameters
+    ----------
+    molecule : qcdb.molecule or psi4.core.Molecule
+        The molecule to compute the frequencies of.
+    irrep : int
+        The Cotton ordered irrep to get frequencies for. Choose -1 for all
+        irreps.
+
+    Returns
+    -------
+    disp_geoms : list of psi4.core.Matrix
+        A list of displaced geometries to compute energies at.
+    """
+    return _geom_generator(molecule, irrep, "2_0")
```

Comment:
I didn't notice any definite violations, but go ahead and run `yapf` over this file (and add `# yapf: disable` (I think)) if it messes up any formatting you like) so `git blame` doesn't get messed up by a future formatting pass.

Reply:
There were two or three lines worth of newline and whitespace corrections, but that's all.

And with that, all of your points should be taken care of. Once the tests pass again, this should be done!